src/quest/manager.py

```diff
@@ -62,17 +62,16 @@ def _remove_workflow(self, workflow_id: str):
         data = next(d for d in self._workflow_data if d[1] == workflow_id)
         self._workflow_data.remove(data)
 
-    def register_serializer(self, obj_type, serializer_func):
-        self._master_serializer.register_serializer(obj_type, serializer_func)
+    def register_serializer(self, obj_type: type, serializer: TypeSerializer[Any]):
+        self._serializer.register_serializer(obj_type, serializer)
 
```

Comment:
Instead of this function, the developer should now provide the type serializers to the MasterSerializer directly, and then provide the MasterSerializer (or the NoopSerializer if desired) to the WorkflowManager constructor directly.
```py
type_sers = {
  MyClass: lambda my_class: ...,
  YourClass: lambda your_class: ...
}
serializer = MasterSerializer(type_sers)
...
workflow_manager = WorkflowManager(..., serializer)
```